Mathlib/NumberTheory/Rayleigh.lean

```diff
@@ -0,0 +1,246 @@
+    have := add_lt_add_of_le_of_lt h₁₂ h₂₂
+    rw [f, ← Int.cast_one] at this
+    simp_rw [← Int.cast_add, Int.cast_lt] at this
+    rwa [← add_assoc, add_lt_add_iff_right, add_right_comm, Int.lt_add_one_iff] at this
+  have := lt_of_le_of_lt h₄ h₃
+  exact (lt_self_iff_false j).1 this
+
+/-- Let `1 < r ∈ ℝ` and `j ∈ ℤ`. Then either `j ∈ B_r` or `B_r` jumps over `j`.
+
+This is unfortunately not in the proof sketch. -/
+theorem hit_or_miss : j ∈ beattySet r ∨ (∃ k : ℤ, k * r < j ∧ j + 1 ≤ (k + 1) * r) := by
+  have hr₀ : r > 0 := lt_trans zero_lt_one hr
+  -- for both cases, the candidate is `k = ⌈(j + 1) / r⌉ - 1`
+  cases lt_or_ge ((⌈(j + 1) / r⌉ - 1) * r) j
+  · refine Or.inr ⟨⌈(j + 1) / r⌉ - 1, by simpa, ?_⟩
+    rw [Int.cast_sub, Int.cast_one, sub_add_cancel, ← div_le_iff hr₀]
+    apply Int.le_ceil
+  · refine Or.inl ⟨⌈(j + 1) / r⌉ - 1, ?_⟩
+    rw [beattySequence, Int.floor_eq_iff]
+    constructor
+    · simpa
+    rw [Int.cast_sub, Int.cast_one, ← lt_div_iff hr₀, sub_lt_iff_lt_add]
+    apply Int.ceil_lt_add_one
+
+/-- Let `1 < r ∈ ℝ` and `j ∈ ℤ`. Then either `j ∈ B'_r` or `B'_r` jumps over `j`.
+
+This is unfortunately not in the proof sketch. -/
+theorem hit_or_miss' : j ∈ beattySet' r ∨ (∃ k : ℤ, k * r ≤ j ∧ j + 1 < (k + 1) * r) := by
+  have hr₀ : r > 0 := lt_trans zero_lt_one hr
+  -- for both cases, the candidate is `k = ⌊(j + 1) / r⌋`
+  cases le_or_gt (⌊(j + 1) / r⌋ * r) j
+  · refine Or.inr ⟨⌊(j + 1) / r⌋, ‹_›, ?_⟩
+    rw [← div_lt_iff hr₀]
+    apply Int.lt_floor_add_one
+  · refine Or.inl ⟨⌊(j + 1) / r⌋, ?_⟩
+    rw [beattySequence', sub_eq_iff_eq_add, Int.ceil_eq_iff, Int.cast_add, Int.cast_one]
+    constructor
+    · rwa [add_sub_cancel]
+    exact sub_nonneg.1 (Int.sub_floor_div_mul_nonneg (j + 1 : ℝ) hr₀)
+
+end Beatty
+
+/-- Generalization of Rayleigh's theorem on Beatty sequences. Let `r` be a real number greater
+than 1. Then every integer is in exactly one of `B_r` or `B'_(r / (r - 1))`. -/
+theorem rayleigh_int {r : ℝ} (hr : r > 1) :
+    ∀ {j : ℤ},
+      (j ∈ beattySet r ∧ j ∉ beattySet' r.conjugateExponent) ∨
+      (j ∉ beattySet r ∧ j ∈ beattySet' r.conjugateExponent) := by
+  intro j
+  by_cases h₁ : j ∈ beattySet r
+  · by_cases h₂ : j ∈ beattySet' r.conjugateExponent
+    · have ⟨k, h₃⟩ := h₁
+      have ⟨m, h₄⟩ := h₂
+      exact (Beatty.no_collision hr ⟨j, ⟨k, h₃⟩, ⟨m, h₄⟩⟩).elim
+    · exact Or.inl ⟨h₁, h₂⟩
+  · by_cases h₂ : j ∈ beattySet' r.conjugateExponent
+    · exact Or.inr ⟨h₁, h₂⟩
+    · have hs := Beatty.s_gt_1 hr
+      have ⟨k, h₁₁, h₁₂⟩ := (Beatty.hit_or_miss hr).resolve_left h₁
+      have ⟨m, h₂₁, h₂₂⟩ := (Beatty.hit_or_miss' hs).resolve_left h₂
+      exact (Beatty.no_anticollision hr ⟨j, k, m, h₁₁, h₁₂, h₂₁, h₂₂⟩).elim
+
+/-- A version of `rayleigh_int` that uses `Real.IsConjugateExponent r s` as a hypothesis. -/
+theorem rayleigh_int' {r s : ℝ} (hr : Real.IsConjugateExponent r s) :
+    ∀ {j : ℤ}, (j ∈ beattySet r ∧ j ∉ beattySet' s) ∨ (j ∉ beattySet r ∧ j ∈ beattySet' s) := by
```

Comment:
Can you instead phrase this in terms of https://leanprover-community.github.io/mathlib4_docs/Mathlib/Init/Set.html#Set.compl

Reply:
Done. The statement is now `(beattySet r)ᶜ = beattySet' s`.